File: ex1_class_cpp/robot.cpp
#include "robot.h"

Robot::Robot(){
	cout<<"Start Robot\n";
}

Robot::~Robot(){
	cout<<"Robot stopped\n";
}

void Robot::move_forward(int distance){
	cout<<"Robot moving forward: "<< distance << "\n";
}

void Robot::move_backward(int distance){
	cout<<"Robot moving backward: "<< distance << "\n";
}

void Robot::move_left(int distance){
	cout<<"Robot moving left: "<< distance << "\n";
}

void Robot::move_right(int distance){
	cout<<"Robot moving right: "<< distance << "\n";
}

File: ex2_class_python/robot.py
''' 
	Example in ROS class
	class Robot.py
'''

class Robot:
	def __init__(self):
			print("Start Robot")

	def move_forward(self, distance):
			print("Robot moving forward: " + str(distance) + " m")

	def move_backward(self, distance):
			print("Robot moving backward: " + str(distance) + " m")

	def move_left(self, distance):
			print("Robot moving left: " + str(distance) + " m")

	def move_right(self, distance):
			print("Robot moving right: " + str(distance) + " m")

	def __del__(self):
			print("Robot stopped")
